### PR TITLE
feat: dashboard perf improvements

### PR DESCRIPTION
**Pull request by YounixM**
Before: 

https://www.loom.com/share/a23eff8890614fd18e918d93e0bc27a7?sid=075fd219-075a-405d-b2b4-96a3abe23a72

After:

https://www.loom.com/share/a10fe5b7dec346768b7edac1a56be8d4?sid=1e4f67a8-24ee-4c97-9092-d894da0dc06c

### CODE REVIEW

**Comment by palashgdev** [comment]
please list down the changes which are fixed

also attach before and after screenshot

**Comment by YounixM** [comment]
@palashgdev @Rajat-Dabade: Please review and approve. We can merge this.

We need to add more unit test cases for variableItem component. Will take it up in our automation effort.

**Review by srikanthccv — APPROVED** [review]
Tested the variable update part. LGTM.

**Comment by srikanthccv** [comment]
@Rajat-Dabade @palashgdev Please review the changes.

File: frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx [hunk 1]
```diff
@@ -38,48 +46,74 @@ function VariableItem({
...
 	const [optionsData, setOptionsData] = useState<(string | number | boolean)[]>(
 		[],
 	);
-	const [isLoading, setIsLoading] = useState<boolean>(false);
+
+	const [variableValue, setVaribleValue] = useState(
+		variableData?.selectedValue?.toString() || '',
+	);
+
+	const debouncedVariableValue = useDebounce(variableValue, 500);
```
Comment (Rajat-Dabade) [inline frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx:59]: No need to pass the second parameter as the default value is `500` inside the function.

**Comment by YounixM** [comment]
@Rajat-Dabade : Will take up test related comments in next PR.

**Review by Rajat-Dabade — APPROVED** [review]
LGTM 🚀

**Comment by YounixM** [comment]
Have marked all conversations as resolved to complete checks. Will take them up in next PR.

**Pull request merged**